### ISSUE TITLE
Error on lines starting with a open parenthesis

### ISSUE THREAD

**Issue by Danmou** [xonsh/xonsh#2515]
Whenever I enter a `(` as the first letter on a line in xonsh, it spews out a traceback like the following:
```
(Exception in thread Thread-112:
Traceback (most recent call last):                                                                      
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner                 
    self.run()                                                                          
  File "/usr/lib/python3.5/threading.py", line 862, in run                              
    self._target(*self._args, **self._kwargs)                                           
  File "/usr/local/lib/python3.5/dist-packages/prompt_toolkit/interface.py", line 860, in run           
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "/usr/local/lib/python3.5/dist-packages/xonsh/ptk/completer.py", line 49, in get_completions
    self.ctx)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 124, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/completers/__amalgam__.py", line 1431, in completee
    python_comps = complete_python(prefix, line, start, end, ctx)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/completers/__amalgam__.py", line 1254, in completen
    rtn = _complete_python(prefix, line, start, end, ctx)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/completers/__amalgam__.py", line 1278, in _completn
    args = python_signature_complete(prefix, line, end, ctx, filt)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/completers/__amalgam__.py", line 1376, in python_se
    sig = inspect.signature(val)
  File "/usr/lib/python3.5/inspect.py", line 2987, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/usr/lib/python3.5/inspect.py", line 2737, in from_callable
    follow_wrapper_chains=follow_wrapped)
  File "/usr/lib/python3.5/inspect.py", line 2156, in _signature_from_callable
    raise TypeError('{!r} is not a callable object'.format(obj))
TypeError: None is not a callable object
```

I'm using xonsh 0.5.12, installed through pip. The problem occurs both with and without autopair enabled. It also occurs when pasting a line starting with a parenthesis, e.g. `(asdf`, unless it also contains the closing parenthesis, e.g. `(asdf)`.

Based on the error message, the problem seems to stem from the tab completion. On further investiagion, I've found that disabling both the `base` and `python` completers (using `completer remove ...`) eliminates the problem, but if either one of them is active I get the error.

Can anyone reproduce the problem or does it happen only on my machine?

**Comment by gforsyth** [comment]
Hey @Danmou -- thanks for reporting.  I can confirm the issue. And yes, it is a problem with tab completion. As a workaround for now, rather than completely disabling the base and python completers, you can set `$UPDATE_COMPLETIONS_ON_KEYPRESS=False` in your `xonshrc`. You will then have to hit TAB to bring up the completions list, but that will stop the error every time you have an unclosed parenthesis (unless you hit tab after the unclosed paren)

This is related to a couple of other issues but I can't find them right now.

**Comment by scopatz** [comment]
Thanks for reporting this @Danmou and tracking this down @gforsyth.  I have put in a potential fix in #2528

**Comment by gforsyth** [comment]
Closed via 65c80f94
